components/admin_console/brand_image_setting/brand_image_setting.jsx

```diff
@@ -66,56 +79,65 @@ export default class BrandImageSetting extends React.PureComponent {
 
     handleImageChange() {
         const element = $(this.refs.fileInput);
-
         if (element.prop('files').length > 0) {
+            this.props.setSaveNeeded();
             this.setState({
                 brandImage: element.prop('files')[0],
-                status: UploadStatuses.DEFAULT,
+                deleteBrandImage: false,
             });
         }
     }
 
-    handleImageSubmit(e) {
-        e.preventDefault();
-
-        if (!this.state.brandImage) {
-            return;
-        }
+    handleDeleteButtonPressed() {
+        this.setState({deleteBrandImage: true, brandImage: null, brandImageExists: false});
+        this.props.setSaveNeeded();
+    }
 
-        if (this.state.status === UploadStatuses.LOADING) {
-            return;
+    handleSave = async () => {
+        if (!this.state.deleteBrandImage && !this.state.brandImage) {
+            return null;
         }
 
         this.setState({
             error: '',
-            status: UploadStatuses.LOADING,
         });
 
-        uploadBrandImage(
-            this.state.brandImage,
-            () => {
-                this.setState({
-                    brandImageExists: true,
-                    brandImage: null,
-                    brandImageTimestamp: Date.now(),
-                    status: UploadStatuses.COMPLETE,
-                });
-            },
-            (err) => {
-                this.setState({
-                    error: err.message,
-                    status: UploadStatuses.DEFAULT,
-                });
-            }
-        );
+        if (this.state.deleteBrandImage) {
+            await deleteBrandImage(
+                () => {
+                    this.setState({
+                        deleteBrandImage: false,
+                        brandImageExists: false,
+                        brandImage: null,
+                    });
+                },
+                (err) => {
+                    this.setState({
+                        error: err.message,
+                    });
+                }
+            );
+        } else if (this.state.brandImage) {
+            await uploadBrandImage(
+                this.state.brandImage,
+                () => {
+                    this.setState({
+                        brandImageExists: true,
+                        brandImage: null,
+                        brandImageTimestamp: Date.now(),
+                    });
+                },
+                (err) => {
+                    this.setState({
+                        error: err.message,
+                    });
+                }
+            );
+        }
+        return this.state.error;
```

Comment:
I don't think this will return an error like you're expecting. `this.setState` doesn't update `this.state` immediately, and in this case, I think it doesn't happen until after we're done responding to the click event that started the submit process.

It would be a bit easier to structure this if it wasn't using these callback-based actions, but you could do something like this to get the error returned:
```javascript
let errorMessage;
if (this.state.deleteImage) {
    await deleteBrandImage(
        () => {...},
        (err) => {
            errorMessage = err.message;
        }
    );
}

return errorMessage;
```

Reply:
Also, it might be better to return an object like `{error: ...}` to indicate that an error occurred since that's more consistent with how we tend to return errors from other async functions

Reply:
Thanks! I've now addressed this.